test/e2e/tests/network/multi-rpc.spec.ts

```diff
@@ -387,93 +348,41 @@ describe('MultiRpc:', function (this: Suite) {
       },
 
       async ({ driver }: { driver: Driver }) => {
-        const password = 'password';
-
-        await driver.navigate();
-
-        await importSRPOnboardingFlow(driver, TEST_SEED_PHRASE, password);
-
-        await driver.delay(regularDelayMs);
-
-        // go to advanced settigns
-        await driver.clickElementAndWaitToDisappear({
-          text: 'Manage default privacy settings',
-        });
-
-        await driver.clickElement({
-          text: 'General',
-        });
-
-        // open edit modal
-        await driver.clickElement({
-          text: 'arbitrum-mainnet.infura.io',
-          tag: 'p',
-        });
-
-        await driver.clickElement('[data-testid="test-add-rpc-drop-down"]');
-
-        await driver.delay(regularDelayMs);
-        await driver.clickElement({
-          text: 'Arbitrum mainnet 2',
-          tag: 'button',
-        });
-
-        await driver.clickElementAndWaitToDisappear({
-          text: 'Save',
-          tag: 'button',
-        });
-
-        await driver.clickElement('[data-testid="category-back-button"]');
-
-        await driver.clickElement(
-          '[data-testid="privacy-settings-back-button"]',
+        await importSRPOnboardingFlow(driver);
+        const onboardingCompletePage = new OnboardingCompletePage(driver);
+        await onboardingCompletePage.check_pageIsLoaded();
+        await onboardingCompletePage.navigateToDefaultPrivacySettings();
+        const onboardingPrivacySettingsPage = new OnboardingPrivacySettingsPage(
+          driver,
         );
+        await onboardingPrivacySettingsPage.check_pageIsLoaded();
+        await onboardingPrivacySettingsPage.navigateToGeneralSettings();
 
-        await driver.clickElementAndWaitToDisappear({
-          text: 'Done',
-          tag: 'button',
-        });
-
-        await driver.clickElement({
-          text: 'Next',
-          tag: 'button',
-        });
-
-        await driver.clickElementAndWaitToDisappear({
-          text: 'Done',
-          tag: 'button',
-        });
-
-        // Validate the network was edited
-        const networkEdited = await driver.isElementPresent({
-          text: '“Arbitrum One” was successfully edited!',
-        });
-        assert.equal(
-          networkEdited,
-          true,
-          '“Arbitrum One” was successfully edited!',
+        // open edit network modal during onboarding and select the second rpc
+        await onboardingPrivacySettingsPage.openEditNetworkModal(
+          'Arbitrum One',
         );
-        // Ensures popover backround doesn't kill test
-        await driver.assertElementNotPresent('.popover-bg');
-
-        // We need to use clickElementSafe + assertElementNotPresent as sometimes the network dialog doesn't appear, as per this issue (#27870)
-        // TODO: change the 2 actions for clickElementAndWaitToDisappear, once the issue is fixed
```

Comment:
This is still flaky even with the changes and has been reported this week https://github.com/MetaMask/metamask-extension/issues/27422. I was unable to reproduce it on my local machine.

Could you add this comment for tracking purposes?

Reply:
the comment was moved  from the test body to the page class method `closeUseNetworkNotificationModal`

Reply:
Perfect @chloeYue! that makes sense too.

Reply:
@chloeYue,Sorry, I changed my thoughts here because this flakiness is related to the test scenario and has nothing to do with the notification model. If you agree could you add in the test scenario itself before clicking this modal and needs to be removed when the bug is fixed.